test/sd-test-srv/main.go

```diff
@@ -0,0 +1,80 @@
+// sd-test-srv runs a simple service discovery system; it returns two hardcoded
+// IP addresses for every A query.
+package main
+
+import (
+	"flag"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+)
+
+func dnsHandler(w dns.ResponseWriter, r *dns.Msg) {
+	m := new(dns.Msg)
+	m.SetReply(r)
+	m.Compress = false
+
+	if len(r.Question) < 1 {
```

Comment:
Probably just want `!= 1` so it properly handles queries with more than one question (unlikely, but so is a query with 0).

Reply:
@jsha Bump on this feedback: was it missed or did you disagree?

Reply:
Thanks for spotting; I missed it.